test/src/unit-iterators2.cpp

```diff
@@ -307,6 +317,7 @@ TEST_CASE("iterators 2")
                     auto it = j_array.begin();
                     it += 3;
                     CHECK((j_array.begin() + 3) == it);
+                    CHECK(json::iterator(3 + j_array.begin()) == it);
```

Comment:
It's a required operation in a random access iterator. Please check: http://en.cppreference.com/w/cpp/concept/RandomAccessIterator